### PR TITLE
Add more fine-grained merge rules for SAB classification

### PR DESCRIPTION
**Pull request by kwahlin**
Rules explained in https://jira.kb.se/browse/FMT-204

### CODE REVIEW

**Review by andand79 — APPROVED** [review]
Très bien!

File: librisworks/src/main/groovy/se/kb/libris/mergeworks/compare/Classification.groovy [hunk 1]
```diff
@@ -79,7 +73,94 @@ class Classification extends StuffSet {
...
+            return b
+        }
+        return null
+    }
+
+    static sabPrecedes(String a, String b) {
+        def (equal, startsWith) = sabPrecedenceRules
+        def startsWithExceptions = ~/^Hcb|^Hdab/
+        def preferred = equal[b] ?: startsWith.find { b.startsWith(it.key) }?.value
+        if (preferred && !(a =~ startsWithExceptions)) {
```
Comment (olovy) [inline librisworks/src/main/groovy/se/kb/libris/mergeworks/compare/Classification.groovy:99]: `startsWithExceptions` also apply to "equals?"
In that case the naming is a bit confusing.

**Review by olovy — CHANGES_REQUESTED** [review]
I think this needs some tests
Also: see comment about startsWithExceptions.

I don't suggest changing the implementation but the code is a bit hard to follow. It could be made clearer by for instance by making it more explicit what the sabPrecedenceRules data structure contains.

**Comment by kwahlin** [comment]
> I think this needs some tests
> 
> I don't suggest changing the implementation but the code is a bit hard to follow. It could be made clearer by for instance by making it more explicit what the sabPrecedenceRules data structure contains.

I agree. I was a bit eager to get this in quickly.

**Comment by olovy** [comment]
> > I think this needs some tests
> > I don't suggest changing the implementation but the code is a bit hard to follow. It could be made clearer by for instance by making it more explicit what the sabPrecedenceRules data structure contains.
> 
> I agree. I was a bit eager to get this in quickly.

Just a couple of tests on `tryMergeSabCodes` or `merge` is fine

**Review by olovy — APPROVED** [review]
LGTM

See comment about `startsWithExceptions`
